proposals/291-loeschkonzept-v2.md

```diff
@@ -0,0 +1,400 @@
+
+Die Löschoperation selbst darf dabei allerdings nur server-lokal und ohne
+direkte Auswirkung auf die Föderation erfolgen damit Löschkonfigurationen auf
+unterschiedlichen Servern nicht interferieren. Diese server-lokale Löschung
+schließt z. B. die Verwendung von Redactions oder das Kicken der Nutzer anderer
+Homeserver aus.
+
+**A_2 - Automatische serverseitige Löschung**
+
+TI-M Pro Fachdienste KÖNNEN Rauminhalte und damit verbundene Medien automatisch
+und ohne Einwilligung der Raumteilnehmer löschen. **\[\<=\]**
+
+**A_3 - Lokale Beschränkung der automatischen serverseitigen Löschung**
+
+Implementieren TI-M Pro Fachdienste Funktionen zum automatisch Löschen von
+Rauminhalten oder damit verbundener Medien, so MUSS die Löschung server-lokal
+und ohne direkten Einfluss auf die Föderation erfolgen. **\[\<=\]**
+
+Damit Inhalte nicht unerwartet verloren gehen, müssen Nutzer über etwaige
+automatische Löschfunktionen zumindest informiert werden. Die konkrete Form der
+Information bleibt dabei dem Anbieter überlassen. So könnte der Anbieter z. B.
+über [Server Notices] einzelne Räumlöschungen vorankündigen. Alternativ wäre es
+auch denkbar, dass nur einmalig über feste Löschintervalle informiert wird.
+
+**A_4 - Information über automatische serverseitige Löschung**
+
+Nutzen TI-M Pro Anbieter Funktionen zur automatischen serverseitigen Löschung
+von Rauminhalte oder damit verbundener Medien, so MÜSSEN sie ihre Nutzer vorab
+darüber informieren. **\[\<=\]**
+
+Die konkrete Ausgestaltung einer automatischen serverseitigen Löschfunktion muss
+durch die Spezifikation nicht weiter vorgegeben werden. Hier gibt es
+verschiedene Möglichkeiten. So könnten z. B. Nutzer des eigenen Homeservers aus
+Räumen entfernt werden und diese Räume samt der zugehörigen Events und Medien
+dann aus der Datenbank des Servers gelöscht werden. Alternativ wäre auch ein
+fortlaufendes Löschen von veralteten Events und Medien aus der Datenbank des
+Servers möglich, wobei der Raum selbst bestehen bleibt. Homeserver wie Synapse
+bieten hierfür konfigurierbare [Message Retention Policies] an. Eine weitere
+Möglichkeit wäre die Implementierung eines Quota-Systems wie es z. B. bei
+E-Mail-Anbietern gängig ist. Die hierfür eventuell notwendige Verlinkung von
+Räumen und Medien könnte beispielsweise durch [MSC3911] realisiert werden.
+
+### Clientseitiges Löschen
+
+**Story 5**
+
+- Als Endnutzer
+- möchte ich nicht mehr benötigte Chats löschen können
+- um Ordnung in meiner Chatliste zu schaffen.
+
+**Story 6**
+
+- Als Mitarbeiter im Gesundheitswesen
+- möchte ich Chats, die ich in ein Drittsystem archiviert habe, löschen
+- um Doppeldokumentation zu vermeiden.
+
+Nutzer müssen für die Organisation ihrer Unterhaltungen in die Lage versetzt
+werden Räume selbstständig verlassen und vom Client entfernen zu können. Hierbei
+ist zu beachten, dass Räume nach erfolgreichem [`/leave`] weiterhin in der
+[`/sync`]-Response auftauchen und dort erst durch ein explizites [`/forget`]
```

Comment:
Kleiner Hinweis: Das ist so standardmäßig aktuell nicht der Fall. Ich zitiere:

> Once a user leaves a room, it will no longer appear in the response to the [/sync](https://spec.matrix.org/v1.13/client-server-api/#get_matrixclientv3sync) API unless it is explicitly requested via a filter with the include_leave field set to true.

Reply:
Oh, du hast recht. Danke! Habe die Formulierung mit https://github.com/gematik/api-ti-messenger/pull/291/commits/cda73cd85247cfe844c6d11a0f3bbeb4f9801f5b angepasst.